### PR TITLE
V22F-741-Linux-Update

### PR DESCRIPTION
**Pull request by KungFuDonkey**
This PR updates the library so that it can be used on Linux.

### CODE REVIEW

File: .github/workflows/code-coverage-linux.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      shell: bash
+      run: |
+        sudo apt update
+        sudo apt install gcc-11 g++-11
+        sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-11 110 --slave /usr/bin/g++ g++ /usr/bin/g++-11 --slave /usr/bin/gcov gcov /usr/bin/gcov-11
+
+    - name: Install GCovr
+      run: |
+        sudo apt update
+        sudo apt install python3-pip
```
Comment (suppergerrie2) [inline .github/workflows/code-coverage-linux.yml:35]: According to the github documentation pip should already be installed (and seems to be the case checking the logs: `python3-pip is already the newest version (20.0.2-5ubuntu1.6).`)

File: CMakeLists.txt [hunk 2]
```diff
@@ -3,13 +3,26 @@
...
-endif(OPEN_CPP_COVERAGE)
+IF(WIN32)
+    find_program(OPEN_CPP_COVERAGE "OpenCppCoverage")
+    if(OPEN_CPP_COVERAGE)
+        MESSAGE(STATUS "OpenCppCoverage -- found")
+    else(OPEN_CPP_COVERAGE)
+        MESSAGE(STATUS "OpenCppCoverage -- not found")
+    endif(OPEN_CPP_COVERAGE)
+ENDIF(WIN32)
+IF(UNIX)
```
Comment (suppergerrie2) [inline CMakeLists.txt:14]: should this be an else if with an else for an unsupported warning?

**Review by suppergerrie2 — CHANGES_REQUESTED** [review]

File: .github/workflows/code-coverage-linux.yml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    - name: Install GCC11
+      shell: bash
+      run: |
+        sudo apt update
+        sudo apt install gcc-11 g++-11
+        sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-11 110 --slave /usr/bin/g++ g++ /usr/bin/g++-11 --slave /usr/bin/gcov gcov /usr/bin/gcov-11
+
+    - name: Install GCovr
+      run: |
+        sudo apt update
```
[the remark this answers is not in the document]
Comment (KungFuDonkey) [reply .github/workflows/code-coverage-linux.yml:34]: linux be like that sometimes, it takes like 0.1 seconds so it doesn't really hurt

File: CMakeLists.txt (hunk 2 above)
Comment (KungFuDonkey) [reply CMakeLists.txt:14]: well I wasnt planning to throw not supported messages as you can read the readme

**Review by suppergerrie2 — APPROVED** [review]

File: IPCTests/SocketTest.cpp [hunk 4]
```diff
@@ -6,7 +6,13 @@
...
 #define AWAIT_CONNECTION_TIMEOUT 3
 
-#define AMOUNT_OF_TESTS 10000
+#ifdef WIN32
+#define DOUBLE_INIT_CODE IPCLIB_SERVER_ERROR
+#endif
+#ifdef __linux__
+#define DOUBLE_INIT_CODE IPCLIB_SUCCEED
+#endif
+#define AMOUNT_OF_TESTS 1
```
Comment (larsdekwant) [inline IPCTests/SocketTest.cpp:15]: You now have for-loops over 1 value, either remove the for-loops or increase the amount of tests.

**Review by larsdekwant — CHANGES_REQUESTED** [review]

File: IPCTests/SocketTest.cpp (hunk 4 above)
Comment (KungFuDonkey) [reply IPCTests/SocketTest.cpp:15]: correct, this was a test. I will change it back to a bigger number

**Review by larsdekwant — APPROVED** [review]

**Pull request merged**